frontend/src/pages/projects/screens/spawner/storage/usePreferredStorageClass.ts

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { AppContext } from '~/app/AppContext';
+import { StorageClassKind } from '~/k8sTypes';
+
+const usePreferredStorageClass = (): StorageClassKind | undefined => {
+  const {
+    dashboardConfig: {
+      spec: { notebookController },
+    },
+    storageClasses,
+  } = React.useContext(AppContext);
+
+  const defaultClusterStorageClasses = storageClasses.filter((storageclass) =>
+    storageclass.metadata.annotations?.['storageclass.kubernetes.io/is-default-class']?.includes(
+      'true',
+    ),
+  );
+
+  const configStorageClassName = notebookController?.storageClassName ?? '';
+
+  if (defaultClusterStorageClasses.length != 0) {
+    return undefined;
+  }
+
+  if (configStorageClassName == '') {
+    return undefined;
+  }
+
+  const storageClassDashBoardConfigVsCluster = storageClasses.filter((storageclass) =>
+    storageclass.metadata.name.includes(configStorageClassName),
```

Comment:
This will give bad results... `foo` and `afoobar` if you search for `foo` it's luck of the draw which one comes first; I actually think it's by default alphabetical sorting, but it shouldn't be left up to this

Reply:
@andrewballantyne no, I think this matches for the complete value, not a partial match

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

returns true or false which is exactly what I am looking for

Reply:
That's an array includes, you're doing a string includes.

This would be the docs you want to look at:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes

Reply:
I could and actually shall do full string compare, ie. something like string.equals
```
const storageClassDashBoardConfigVsCluster = storageClasses.filter((storageclass) =>
    storageclass.metadata.name === configStorageClassName,
```